packages/ui/src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import {
+  forwardRef,
+  HTMLChakraProps,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput as ChakraNumberInput,
+  NumberInputField,
+  NumberInputStepper,
+  UseNumberInputProps,
+} from '@chakra-ui/react';
+
+type NumberInputSize = 'sm' | 'md' | 'lg';
+
+type NumberInputMode = 'numeric' | 'decimal';
+
+export interface NumberInputProps extends Omit<HTMLChakraProps<'div'>, keyof UseNumberInputProps> {
+  size: NumberInputSize;
+  inputMode: NumberInputMode;
+  disabled: boolean;
+  value?: number;
+  onChange?: (arg: string) => void;
+  max?: number;
+  min?: number;
+  precision?: number;
+  step?: number;
+}
+
+export const lgStyleProps = {
+  height: '48px',
+  padding: '1px 0px 1px 16px',
+};
+
+export const mdStyleProps = {
+  height: '40px',
+  padding: '1px 0px 1px 16px',
+};
+
+export const smStyleProps = {
+  height: '32px',
+  padding: '1px 0px 1px 12px',
+};
+
+export const NumberInput = forwardRef<NumberInputProps, 'input'>(
+  (
+    {
+      size,
+      inputMode,
+      disabled,
+      'aria-describedby': ariaDescribedby,
+      'aria-label': ariaLabel,
+      'aria-labelledby': ariaLabelledby,
+      value,
+      name,
+      max,
+      min,
+      precision,
+      onChange,
+      step,
```

Comment:
You don't have to destructure these props, just spread the rest of them like: 
`...props`
and then pass it to the chakra component

Reply:
I tried do this like you recommend but I got an error on ChakraNumberInput about invalid type of props. So I decided to destructure this props

Reply:
If you can, move back to the version with spread and paste an error you're getting. Destructuring so many props makes this component a little harder to read. Of course, if there is no other way we will leave it, but it's still worth trying to make it cleaner.